examples/private_set_intersection/module/rust/src/lib.rs

```diff
@@ -49,32 +49,46 @@ oak::entrypoint!(oak_main => |_in_channel| {
 pub const SET_THRESHOLD: u64 = 2;
 
 #[derive(Default)]
-struct Node {
+struct SetIntersection {
     /// Current intersection of contributed private sets.
-    values: Option<HashSet<String>>,
+    values: HashSet<String>,
     /// Number of contributed private sets.
     set_count: u64,
     /// The intersection is locked and new contributions are discarded.
     locked: bool,
 }
 
+#[derive(Default)]
+struct Node {
+    /// Map from set ID to `SetIntersection`.
+    sets: HashMap<String, SetIntersection>,
+}
+
 impl PrivateSetIntersection for Node {
     fn submit_set(&mut self, req: SubmitSetRequest) -> grpc::Result<()> {
-        if self.locked {
+        let mut current_set = self
+            .sets
+            .entry(req.set_id)
+            .or_insert(SetIntersection::default());
+
+        if current_set.locked {
             return Err(grpc::build_status(
                 grpc::Code::FailedPrecondition,
                 "Set contributions are no longer available",
             ));
         }
 
-        if self.set_count < SET_THRESHOLD {
-            let set = req.values.iter().cloned().collect::<HashSet<_>>();
-            let next = match self.values {
-                Some(ref previous) => previous.intersection(&set).cloned().collect(),
-                None => set,
+        if current_set.set_count < SET_THRESHOLD {
+            let submitted_set = req.values.iter().cloned().collect::<HashSet<_>>();
+            current_set.values = if current_set.values.is_empty() {
+                submitted_set
+            } else {
+                current_set
+                    .values
+                    .intersection(&submitted_set)
+                    .cloned()
+                    .collect()
```

Comment:
Where is the code that increments the sample count now?

If it's missing, how can it work?

Reply:
Done